src/firewall/core/nftables.py

```diff
@@ -411,20 +412,19 @@ def build_flush_rules(self):
 
         return self._build_delete_table_rules(TABLE_NAME)
 
-    def _build_set_policy_rules_ct_rules(self, enable):
+    def _build_set_policy_rules_ct_rules(self, enable, hook):
         add_del = { True: "add", False: "delete" }[enable]
-        rules = []
-        for hook in ["input", "forward", "output"]:
-            rules.append({add_del: {"rule": {"family": "inet",
-                                             "table": TABLE_NAME_POLICY,
-                                             "chain": "%s_%s" % ("filter", hook),
-                                             "expr": [{"match": {"left": {"ct": {"key": "state"}},
-                                                                 "op": "in",
-                                                                 "right": {"set": ["established", "related"]}}},
-                                                      {"accept": None}]}}})
-        return rules
-
-    def build_set_policy_rules(self, policy):
+        return [
+            {add_del: {"rule": {"family": "inet",
+                                         "table": TABLE_NAME_POLICY,
+                                         "chain": "%s_%s" % ("filter", hook),
+                                         "expr": [{"match": {"left": {"ct": {"key": "state"}},
+                                                             "op": "in",
+                                                             "right": {"set": ["established", "related"]}}},
+                                                  {"accept": None}]}}}
+        ]
+
+    def build_set_policy_rules(self, policy, drop_details):
```

Comment:
This refactor looks unnecessary. I presume it's because you're trying to avoid the rule if the ReloadPolicy is ACCEPT.. but that's not worth the complexity. Just keep the rule in all scenarios.

Reply:
Which part looks unnecessary? The loop in `build_set_policy_rules()` needs to iterate over the hooks (`("INPUT", "FORWARD", "OUTPUT")`), as the policy is per-hook. What is your exact suggestion?

(I adjusted this part of the patch somewhat, that `_build_set_policy_rules_ct_rule()` now only returns one rule and not a list of rules. But the gist is the same).